### PR TITLE
Add comments and refactor style

### PR DESCRIPTION
**Pull request by dmsnell**
While working on tracking down a nasty synchronization bug
I could keep all the data flows and puzzle pieces together
in my mind, so I wrote some of them down as best as I was
understanding them.

Further I refactored some code because I find the terser
styles afforded by ES2015+ syntax to confuse me less,
especially when following around closures with `bind` and
`call` and `apply`

The point isn't to change behaviors here but to figure them
out and write them down. That is, there should be no
functional changes in this patch and if there are then it's
probably a bug.

### CODE REVIEW

File: src/simperium/channel.js [hunk 1]
```diff
@@ -47,6 +54,7 @@ internal.buildModifyChange = function( id, object, ghost ) {
 	}
 
 	if ( empty ) return this.emit( 'unmodified', id, object, ghost );
+	payload.v && payload.v.content && console.log(payload.v.content)
```
Comment (dmsnell) [inline src/simperium/channel.js]: oops will take this out - part of my investigation

File: src/simperium/channel.js [hunk 2]
```diff
@@ -275,7 +376,9 @@ export default function Channel( appid, access_token, bucket, store ) {
 			options = { sync: true };
 		}
 
+		console.time('update')
 		return update.call( bucket, id, object, options, function( err, object ) {
+			console.timeEnd('update')
```
Comment (dmsnell) [inline src/simperium/channel.js]: will remove

**Comment by beaucollins** [comment]
I've been exploring flowtype for this project and have cleaned up the Channel module quite a bit while experimenting. Take a look at #60 to see a much better documented and cleaner interface for [`src/simperium/channel.js`](https://github.com/Simperium/node-simperium/blob/9d14a1545ffd34d19f504351a322dcdbc03c97a4/src/simperium/channel.js)

**Comment by dmsnell** [comment]
Not enough time to rebase this and update it. It was a helpful exploration and hopefully it will happen again in the future.

**Pull request closed without merging**